### PR TITLE
Fix compilation incomplete type compilation error due to forward declare

### PR DESCRIPTION
**Pull request by piiertho**
This needs to be tested.
I got it compiling on linux with those changes (still does not compile on windows)
Forward declarations were causing troubles, and protobuf was not linking.

### CODE REVIEW

File: src/jni/env.h [hunk 1]
```diff
@@ -41,6 +41,13 @@ namespace jni {
         friend class JArray;
         friend class JObjectArray;
     };
+
+    template <class T>
+    inline T JObject::new_global_ref(Env& env) {
```
Comment (piiertho) [inline src/jni/env.h:46]: Not sure that's a good way to solve the error

File: SCsub [hunk 2]
```diff
@@ -13,8 +13,12 @@ protobuf_include_dir = [
 ]
 
 protobuf_lib_dir = Dir("libprotobuf/lib").path
-env.Append(LINKFLAGS="-L" + protobuf_lib_dir)
-env.Append(LIBS="protobuf")
+if env["platform"] == "windows":
+    env.Append(LINKFLAGS="-LIBPATH:" + protobuf_lib_dir)
```
Comment (raniejade) [inline SCsub:17]: ohhh, so it's a `-` not `/`? The microsoft docs says it's a slash XD

File: src/kt_variant.h [hunk 3]
```diff
@@ -3,6 +3,10 @@
 #include "wire/wire.pb.h"
 #include "core/variant.h"
 
+// must match the value order of godot_variant_type
+static void(*TO_KT_VARIANT_FROM[27 /* Variant::Type count */])(wire::Value&, const Variant&);
```
Comment (raniejade) [inline src/kt_variant.h:7]: does it need to be in the header? I mean clients of the class shouldn't really know about this arrays.

**Review by raniejade — APPROVED** [review]

File: SCsub (hunk 2 above)
Comment (piiertho) [reply SCsub:17]: It replace it by /, do you want I put / instead ?

File: src/kt_variant.h (hunk 3 above)
Comment (piiertho) [reply src/kt_variant.h:7]: yep that's true, let me change this.

File: SCsub (hunk 2 above)
Comment (raniejade) [reply SCsub:17]: Yeah, I think it's better to use `/`.

Comment (piiertho) [reply SCsub]: done.

**Pull request merged**